fix: #35 - added globalErrorHandling and resolved issue

Jainil2004 opened the pull request:
## issue #35: resolved

**Related Issues:**
Issue #35: Global Exception Handling 

Fixes #35 

**Description:**
1. added a new middleware: globalErrorHandlingMiddleware.js which contains the logic for handling any sort of unhandled errors encountered during the request-response cycle. the middleware will log the details and send an 'error' response to the client.
2. modified app.js: the modification is made so that the globalErrorHandlingMiddleware works perfectly at times required. And is able to catch exceptions during the request and response cycle.
 

**Checklist:**
- [x] I have tested my changes.
- [x] My code follows the project's coding standards.
- [x] I have updated the documentation (if applicable).

**Screenshots:**
1. middleware: 
![image](https://github.com/AnupK1234/MealMatters/assets/112818722/16de981b-ea1e-4161-8c67-d32bf09be9b8)
2. app.js:
![image](https://github.com/AnupK1234/MealMatters/assets/112818722/4d0313ca-f65f-4be2-8b76-232e2a0e1b5c)
![image](https://github.com/AnupK1234/MealMatters/assets/112818722/05c86bef-338c-4a31-95de-ea8a83747c22)



**Additional information** <!-- Optional -->
The middleware does handle the connection errors. but the response and logging can be improved to exactly pinpoint where the problem is for future debugging. At the present moment it only tells that something went wrong during the request-response cycle. but doesn't actually tell what actually happened and where exactly it happened.

AnupK1234 reviewed (changes requested):
Try to handle atleast one type of error.

Jainil2004 commented:
issue noted. I will add handling for mongoDB errors. which would be a specific type of error requested

Jainil2004 commented:
@AnupK1234 I've removed the global error handling functionality and added a specific error handling middleware (MongoError). Please have a look again and let me know if more changes are to be made.
